File: ActivityApp/ActivityApp/ViewController.m
//
//  ViewController.m
//  ActivityApp
//
//  Created by Raunak Talwar on 8/5/15.
//  Copyright (c) 2015 Raunak Talwar. All rights reserved.
//

#import "ViewController.h"
#import "EBCustomCell.h"
#import "EBActivity.h"
#import "EBParser.h"
#import "ActivityTableViewController.h"
#import "ActivityCollectionViewController.h"
@interface ViewController ()


@property (weak, nonatomic) IBOutlet UIButton *tableViewButton;

@property (weak, nonatomic) IBOutlet UIButton *collectionViewButton;
@end

@implementation ViewController
- (IBAction)tableViewButtonPressed:(id)sender {
    
    NSLog(@"Table View Button Pressed");
    ActivityTableViewController *viewController = [[ActivityTableViewController alloc] init];
   
    [self presentViewController:viewController animated:YES completion:nil];
    
    
    
//    [self.navigationController pushViewController:viewController animated:YES];

}

- (IBAction)collectionViewButtonPressed:(id)sender {

    NSLog(@"Collection View Button Pressed");
    
    ActivityCollectionViewController *viewController = [[ActivityCollectionViewController alloc]init];
    [self.navigationController pushViewController:viewController animated:YES];
    

    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
        
    
}


- (NSTimeInterval)transitionDuration:(id <UIViewControllerContextTransitioning>)transitionContext{
    return 2;
}


- (void)animateTransition:(id <UIViewControllerContextTransitioning>)transitionContext{
    
    
    UIViewController *fromVC=[transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
    UIViewController *toVC=[transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];
    UIView *contextView=[transitionContext containerView];
    
    [contextView insertSubview:toVC.view belowSubview:fromVC.view];
    
    [UIView animateWithDuration:[self transitionDuration:transitionContext] animations:^{
        fromVC.view.alpha=0;
        toVC.view.alpha=1;
    }completion:^(BOOL finished) {
        [transitionContext completeTransition:![transitionContext transitionWasCancelled]];
    }];
    
}




- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (BOOL)prefersStatusBarHidden
{
    return YES;
}

@end

File: SwiftApp1/SwiftApp1/ViewController.swift
//
//  ViewController.swift
//  SwiftApp1
//
//  Created by Raunak Talwar on 8/20/15.
//  Copyright (c) 2015 Raunak Talwar. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
       var myString1 = "hi"
        println(myString1)
        
        var varA = 42
        println(varA)
    
        var varB:Float
        varB = 53.5
        println(varB)
        
        var 你好 = "你好世界"
        println(你好)
        
        println("hi the value of varB is \(varB)")
        
        //Using Optional Variables 
        var myString:String!
        
        myString = "Hello, Swift!"
        
        if myString != nil {
            println( myString )
        }else{
            println("myString has nil value")
        }
        if let yourString = myString {
            println("Your string has - \(yourString)")
        }else{
            println("Your string does not have a value")
        }
        //Using Constants 
        let constA = 42// 
        println(constA)
        
    
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

